lms/djangoapps/certificates/api.py

```diff
@@ -849,7 +849,7 @@ def _course_uses_available_date(course):
     )
 
 
-def available_date_for_certificate(course, certificate):
```

Comment:
I added a fair amount of type checking which you will see  here and below, because it became a side effect of my understanding what the methods did so I  could understand them enough to safely modify or remove.